contracts/account/modules/P256Validator.sol

```diff
@@ -0,0 +1,80 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity ^0.8.24;
+
+import {IERC1271} from "../../interfaces/IERC1271.sol";
+import {PackedUserOperation} from "../../interfaces/IERC4337.sol";
+import {P256} from "../../utils/cryptography/P256.sol";
+import {IERC7579Validator, MODULE_TYPE_VALIDATOR} from "../../interfaces/IERC7579Module.sol";
+import {ERC1271TypedSigner} from "../ERC1271TypedSigner.sol";
+import {ERC4337Utils} from "../utils/ERC4337Utils.sol";
+
+abstract contract P256Validator is ERC1271TypedSigner, IERC7579Validator {
+    mapping(address => bytes32) private _associatedQx;
+    mapping(address => bytes32) private _associatedQy;
+
+    event P256SignerAssociated(address indexed account, bytes32 qx, bytes32 qy);
+    event P256SignerDisassociated(address indexed account);
+
+    function isModuleType(uint256 moduleTypeId) public pure virtual returns (bool) {
+        return moduleTypeId == MODULE_TYPE_VALIDATOR;
+    }
+
+    /// @inheritdoc IERC7579Validator
+    function validateUserOp(
+        PackedUserOperation calldata userOp,
+        bytes32 userOpHash
+    ) public view virtual returns (uint256) {
+        return
+            _validateSignature(userOpHash, userOp.signature)
+                ? ERC4337Utils.SIG_VALIDATION_SUCCESS
+                : ERC4337Utils.SIG_VALIDATION_FAILED;
+    }
+
+    /// @inheritdoc IERC7579Validator
+    function isValidSignatureWithSender(
```

Comment:
This should have the mitigations from ERC1271TypedSigner.

On the other hand, ERC1271TypedSigner defines an external `isValidSignature` function that is not needed for a validator module.